GliaWidgets/Sources/Download/FileDownload.swift

```diff
@@ -90,18 +90,27 @@ class FileDownload {
                 )
                 self.storage.store(data.data, for: storageID)
                 self.state.value = .downloaded(file)
-            } else if let error = error {
+            } else if let error = error as? CoreSdkClient.GliaCoreError {
                 self.state.value = .error(Error(with: error))
+            } else if error != nil {
+                self.state.value = .error(.generic)
             }
         }
 
         state.value = .downloading(progress: progress)
 
-        environment.fetchFile(
+        environment.fetchFile.fetchFile(
```

Comment:
Unfortunate naming, don't know if there is something that can be done.

Reply:
Renamed it c00d78d61d6bba81d352ea7c53db5b6b6350b5be.